src/lib/Libutil/pbs_aes_encrypt.c

```diff
@@ -54,7 +56,30 @@ pbs_encrypt_data(char *uncrypted, int *credtype, size_t len, char **crypted, siz
                 return -1;
 
         EVP_CIPHER_CTX_cleanup(&ctx);
+#else
+        EVP_CIPHER_CTX *ctx;
+
+        ctx = EVP_CIPHER_CTX_new();
+
+        if (ctx == NULL)
+                return -1;
+
+        if (EVP_EncryptInit_ex(ctx, EVP_aes_256_cbc(), NULL, (const unsigned char *) pbs_aes_key, (const unsigned char *) pbs_aes_iv) == 0)
+                return -1;
+
+        plen = len + EVP_CIPHER_CTX_block_size(ctx) + 1;
+        cblk = malloc(plen);
+        if (!cblk)
+                return -1;
+
+        if (EVP_EncryptUpdate(ctx, cblk, &plen, uncrypted, len) == 0)
+                return -1;
+
+        if (EVP_EncryptFinal_ex(ctx, cblk + plen, &len2) == 0)
+                return -1;
 
+        EVP_CIPHER_CTX_free(ctx);
+#endif
```

Comment:
You should be able to reduce some duplicate code by using an approach like this...

```
#if OPENSSL_VERSION_NUMBER < 0x10100000L
#define CIPHER_CONTEXT_INIT(v) EVP_CIPHER_CTX_init(v)
#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_cleanup(v)
    EVP_CIPHER_CTX real_ctx;
    EVP_CIPHER_CTX *ctx = &real_ctx;
#else
#define CIPHER_CONTEXT_INIT(v) v = EVP_CIPHER_CTX_new(); if (!v) return -1;
#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_free(v)
    EVP_CIPHER_CTX *ctx = NULL;
#endif
    CIPHER_CONTEXT_INIT(ctx);
    /* Common encryption code. */
    CIPHER_CONTEXT_CLEAN(ctx);
```